process.go

```diff
@@ -0,0 +1,179 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"io"
+
+	"github.com/urfave/cli"
+)
+
+// == Direct input processing ========================================
+
+func processInputFiles(c *cli.Context, g *Gomplate) error {
+	inputs, err := readInputs(c.String("in"), c.StringSlice("file"))
+	if err != nil {
+		return err
+	}
+
+	outputs := c.StringSlice("out")
+	if len(outputs) == 0 {
+		outputs = []string{"-"}
+	}
+
+	for n, input := range inputs {
+		if err := renderTemplate(g, input, outputs[n]); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// == Recursive input dir processing ======================================
+
+func processInputDir(c *cli.Context, g *Gomplate) error {
+	inputDir := c.String("input-dir")
+	outDir := c.String("output-dir")
+	if _, err := assertDir(outDir); err != nil {
+		return err
+	}
+	return processDir(g, inputDir, outDir)
+}
+
+func processDir(g *Gomplate, inPath string, outPath string) error {
+	inPath = filepath.Clean(inPath)
+	outPath = filepath.Clean(outPath)
+
+	// assert tha input path exists
+	si, err := assertDir(inPath)
+	if err != nil {
+		return err
+	}
+
+	// ensure output directory
+	if err = ensureDir(outPath, si.Mode()); err != nil {
+		return err
+	}
+
+	// read directory
+	entries, err := ioutil.ReadDir(inPath)
+	if err != nil {
+		return err
+	}
+
+	// process or dive in again
+	for _, entry := range entries {
+		nextInPath := filepath.Join(inPath, entry.Name())
+		nextOutPath := filepath.Join(outPath, entry.Name())
+
+		if entry.IsDir() {
+			err := processDir(g, nextInPath, nextOutPath)
+			if err != nil {
+				return err
+			}
+		} else {
+			inString, err := readInput(nextInPath)
+			if err != nil {
+				return err
+			}
+			if err := renderTemplate(g, inString, nextOutPath); err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+// == Rendering ====================================================
+
+func renderTemplate(g *Gomplate, inString string, outPath string) error {
```

Comment:
I think I'd rather this function live in `main.go`. This file seems like a pretty good place to keep all of the file I/O functions, but this one doesn't _really_ fit.